docs/manual/other-topics/sub-queries.md

```diff
@@ -0,0 +1,164 @@
+# Sub Queries
+
+Consider you have two models, `Post` and `Reaction`, with a One-to-Many relationship set up, so that one post has many reactions:
+
+```js
+const Post = sequelize.define('post', {
+    content: DataTypes.STRING
+}, { timestamps: false });
+
+const Reaction = sequelize.define('reaction', {
+    type: DataTypes.STRING
+}, { timestamps: false });
+
+Post.hasMany(Reaction);
+Reaction.belongsTo(Post);
+```
+
+*Note: we have disabled timestamps just to have shorter queries for the next examples.*
+
+Let's fill our tables with some data:
+
+```js
+async function makePostWithReactions(content, reactionTypes) {
+    const post = await Post.create({ content });
+    await Reaction.bulkCreate(
+        reactionTypes.map(type => ({ type, postId: post.id }))
+    );
+    return post;
+}
+
+await makePostWithReactions('Hello World', [
+    'Like', 'Angry', 'Laugh', 'Like', 'Like', 'Angry', 'Sad', 'Like'
+]);
+await makePostWithReactions('My Second Post', [
+    'Laugh', 'Laugh', 'Like', 'Laugh'
+]);
+```
+
+Now, we are ready for examples of the power of subqueries.
+
+Let's say we wanted to compute via SQL a `laughReactionsCount` for each post. We can achieve that with a sub-query, such as the following:
+
+```sql
+SELECT
+    *,
+    (
+        SELECT COUNT(*)
+        FROM reactions AS reaction
+        WHERE
+            reaction.postId = post.id
+            AND
+            reaction.type = "Laugh"
+    ) AS laughReactionsCount
+FROM posts AS post
+```
+
+If we run the above raw SQL query through Sequelize, we get:
+
+```json
+[
+  {
+    "id": 1,
+    "content": "Hello World",
+    "laughReactionsCount": 1
+  },
+  {
+    "id": 2,
+    "content": "My Second Post",
+    "laughReactionsCount": 3
+  }
+]
+```
+
+So how can we achieve that with more help from Sequelize, without having to write the whole raw query by hand?
+
+The answer: by combining the `attributes` option of the finder methods (such as `findAll`) with the `sequelize.literal` utility function, that allows you to directly insert arbitrary content into the query without any automatic escaping.
+
+This means that Sequelize will help you with the main, larger query, but you will still have to write that sub-query by yourself:
+
+```js
+Post.findAll({
+    attributes: {
+        include: [
+            [
+                // Note the wrapping parentheses in the call below!
+                sequelize.literal(`(

```

Comment:
Hmm, include support this? Do we have any tests for this case?

Reply:
Good catch. We don't have any tests for this currently. However, it does work!! Should I just add one test for this? Or do you think this shouldn't be officially supported for some reason? I think it's great that it works :)

Reply:
Keep it